controller/configmap/configmapcontroller.go

```diff
@@ -0,0 +1,171 @@
+package configmap
+
+/*
+Copyright 2020 Crunchy Data Solutions, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"github.com/crunchydata/postgres-operator/config"
+	pgoinformers "github.com/crunchydata/postgres-operator/pkg/generated/informers/externalversions/crunchydata.com/v1"
+	pgolisters "github.com/crunchydata/postgres-operator/pkg/generated/listers/crunchydata.com/v1"
+
+	log "github.com/sirupsen/logrus"
+
+	apiv1 "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const workerCount = 2
```

Comment:
This represents the number of workers created that listen for items on the ConfigMap controller's worker queue, allowing for concurrent processing of items off the queue (and therefore concurrent ConfigMap syncs).  I considered this an internal implementation detail, with the design allowing each controller to individually define a it's own number workers (I would like to look at the other controllers more closely before introducing more workers, so right now only the ConfigMap controller has multiple workers, with all others using a single worker).

To put some additional context behind this, here is where the Kube `Deployment` controller is defaulting it's number of workers:

https://github.com/kubernetes/kubernetes/blob/1b57bd7e1fce60d0d99fd3163b68adbbe05cd3fb/pkg/controller/deployment/config/v1alpha1/defaults.go#L37-L39

Anyway, while I initially viewed this as an implementation detail, we could def expose these setting to allow for fine tuning.  For instance, at scale it could be desirable to have more workers (I could add these as `pgo` config settings within the `pgo.yaml`; like the refresh intervals).

Anyway, just let me know what direction you'd like to go with this one.  I can see the benefit of exposing things like this for administrative fine-tuning (while also keeping them out of the default `pgo.yaml`, since they hopefully will not require tuning for the typical use case).

And either way I will go ahead and add a comment describing the `const`.

Reply:
I would opt to make this tunable, but perhaps not for this release. For instance, why would this be 2, and the others 1? We need more info IMV.

Reply:
Cool, that sounds good.  I think these settings will ultimately be tricky/impossible to nail down to a one-size-fits-all answer, so having an easy way to configure them does provide an easy/convenient way for tuning them to try to find a sweet spot.

And the main reason I didn't bump up the worker counts for the others were simply because I felt that change should be independently made and verified per-controller to ensure no side effects.  In theory it shouldn't be an issue since per the workqueue docs "a single item will not be processed multiple times concurrently" (https://godoc.org/k8s.io/client-go/util/workqueue), but since the various other controllers were not designed with this in mind (as the ConfigMap controller was), I wanted to proceed with caution before making any wholesale changes to worker counts.  And for any controllers that typically have a low volume multiple workers might now even be needed (and the other side of this are the controllers that don't currently have worker queues at all, but should have them).

If we do make them configurable now it will provide an easy way to start playing with these settings, so I can honestly go either way with this one at this point...

Reply:
If you think you can make them configurable in fairly short order... :wink:

Reply:
The various worker counts are now configurable via the `pgo.yaml`.